### PR TITLE
05 hash table

### LINKED ISSUE #12: Hash Algorithms
Proposed Hash Algo
1. input the string argument n
2. find the ascii value of each character in n
3. add the ascii values as value
4. return the remainder of value divided by 1000 

Hash function example:
```
Start
input "Raymond"
R = 82, a = 97, y = 121, m = 109, o = 111, n = 110, d = 100
82 + 97 + 121 + 109 + 111 + 110 + 100 = 730
return 730 % 1000 = 730
End
```

See code stub below
```py
def awesome_hash_func(value=None):
    # do awesome stuff here
    return awesome_hash_value

hash_value = awesome_hash_func("raymond")
print(hash_value) # prints 730
```

A hash table has some major operations which are `get`, `set`, `delete`, `has`, `hash`, `get_keys`

The tests should look something like this
```py
hash_table = HashTable()
value = hash_table.set("Raymond")
self.assertEqual(hash_table.value.get("Raymond"), 730)
self.assertEqual(hash_table.get_keys(), {730: "Raymond"})
```

### PR DESCRIPTION
**Pull request by ebosetalee**
Resolves #12 
Closes #12 

Kindly check the tests and function's arrangement in hash_table
Any corrections? @jirevwe

### CODE REVIEW

**Review by jirevwe — CHANGES_REQUESTED** [review]
You test class should have more test cases

**Comment by ebosetalee** [comment]
> You test class should have more test cases

added more tests. Are they all necessary? @jirevwe

File: hash_table/test/hash_table_test.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    def test_has_in_dict(self):
+        pass
+
+    def test_has_not_in_dict(self):
+        pass
+
+    def test_has_collision(self):
+        pass
+
+    def test_delete_not_in_list(self):
```
Comment (jirevwe) [inline hash_table/test/hash_table_test.py]:
```suggestion
    def test_delete_not_in_hash_table(self):
```

File: hash_table/test/hash_table_test.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        pass
+
+    def test_delete_in_list(self):
```
Comment (jirevwe) [inline hash_table/test/hash_table_test.py]:
```suggestion
    def test_delete_in_hash_table(self):
```

File: hash_table/test/hash_table_test.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    def test_empty_hash_table(self):
+        hash_table = HashTable()
+        self.assertIsNone(hash_table.get_keys())
+
+    def test_hash(self):
+        hash_table = HashTable()
+        value = "Jasmine Smith"
+        self.assertEqual(hash_table.hash(value), 260)
+
+    def test_get_not_in_dict(self):
```
Comment (jirevwe) [inline hash_table/test/hash_table_test.py]:
```suggestion
    def test_get_not_in_hash_table(self):
```

**Review by jirevwe — CHANGES_REQUESTED** [review]
The data structure is not a `list` or a `dict`. Change all the occurrences.

File: hash_table/test/hash_table_test.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    def get_in_dict(self):
+        pass
+
+    def test_set_empty_dict(self):
+        pass
+
+    def test_set_not_empty_dict(self):
+        pass
+
+    def test_set_collision(self):
```
Comment (jirevwe) [inline hash_table/test/hash_table_test.py]:
```suggestion
    def test_set_collision_occurred(self):
```

**Review by jirevwe — CHANGES_REQUESTED** [review]

**Review by jirevwe — COMMENTED** [review]
Love eet

**Comment by ebosetalee** [comment]
@Jirevwe 
done

**Review by yudori — APPROVED** [review]
Welldone!

**Review by jirevwe — APPROVED** [review]

**Pull request merged**